pkg/controller/nodeconfig/status.go

```diff
@@ -4,53 +4,17 @@ package nodeconfig
 
 import (
 	"context"
-	"fmt"
 
 	scyllav1alpha1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1alpha1"
-	"github.com/scylladb/scylla-operator/pkg/controllerhelpers"
-	"github.com/scylladb/scylla-operator/pkg/naming"
-	appsv1 "k8s.io/api/apps/v1"
-	corev1 "k8s.io/api/core/v1"
 	apiequality "k8s.io/apimachinery/pkg/api/equality"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/klog/v2"
 )
 
-func (ncc *Controller) calculateStatus(nc *scyllav1alpha1.NodeConfig, daemonSets map[string]*appsv1.DaemonSet, scyllaUtilsImage string) (*scyllav1alpha1.NodeConfigStatus, error) {
+func (ncc *Controller) calculateStatus(nc *scyllav1alpha1.NodeConfig) (*scyllav1alpha1.NodeConfigStatus, error) {
 	status := nc.Status.DeepCopy()
 	status.ObservedGeneration = nc.Generation
 
-	// TODO: We need to restructure the pattern so status update already know the desired object and we don't
-	// 		 construct it twice.
-	requiredDaemonSet := makeNodeSetupDaemonSet(nc, ncc.operatorImage, scyllaUtilsImage)
-	ds, found := daemonSets[requiredDaemonSet.Name]
-	if !found {
-		klog.V(4).InfoS("Existing DaemonSet not found", "DaemonSet", klog.KObj(ds))
-		return status, nil
-	}
-
-	reconciled, err := controllerhelpers.IsDaemonSetRolledOut(ds)
-	if err != nil {
-		return nil, fmt.Errorf("can't determine is a daemonset %q is reconiled: %w", naming.ObjRef(ds), err)
-	}
-
-	cond := scyllav1alpha1.NodeConfigCondition{
-		Type:               scyllav1alpha1.NodeConfigReconciledConditionType,
-		ObservedGeneration: nc.Generation,
-	}
```

Comment:
Where it is being set now? I don't see it in PR.

Reply:
it isn't - progressing and available conditions are used instead: https://github.com/scylladb/scylla-operator/pull/1918/files?diff=split&w=0#diff-fe70114ecaf255f5bde24d205c100d57e5b6417be893421752f55453b2cb4bedR60-R72

Reply:
then your should raise the apiversion to v1alpha2

Reply:
I'll add it back, although the tests won't be depending on it

Reply:
done

Reply:
@tnozicka is there any benefit in doing this over what I did with computing this from available/progressing/degraded conditions?

Reply:
I haven't seen the new projection at a time I wrote this - looking at it, it changes the semantics of it but I guess that's ok